implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/PrometheusMeterRegistry.java

```diff
@@ -591,11 +590,11 @@ private void applyToCollector(Meter.Id id, Consumer<MicrometerCollector> consume
 
             meterRegistrationFailed(id,
                     "Prometheus requires that all meters with the same name have the same"
-                            + " set of tag keys. There is already an existing meter named '" + id.getName()
+                            + " set of tag keys. There is already an existing meter named '" + getConventionName(id)
                             + "' containing tag keys ["
                             + String.join(", ", collectorMap.get(getConventionName(id)).getTagKeys())
-                            + "]. The meter you are attempting to register" + " has keys ["
-                            + getConventionTags(id).stream().map(Tag::getKey).collect(joining(", ")) + "].");
+                            + "]. The meter you are attempting to register" + " has keys [" + String.join(", ", tagKeys)
+                            + "].");
```

Comment:
This makes the message to use `_`, previously it was using `.`; with the 0.x client we are using them mixed (name uses `.`, tags use `_`).

So this will be something like:
```
[...] There is already an existing meter named 'my_counter' containing tag keys [test_k1]. The meter you are attempting to register has keys [test_k2]."
```